crates/typst-library/src/model/bibliography.rs

```diff
@@ -192,6 +193,48 @@ impl BibliographyElem {
     }
 }
 
+impl Packed<BibliographyElem> {
+    /// Produces the heading for the bibliography, if any.
+    pub fn realize_title(&self, styles: StyleChain) -> Option<Content> {
+        let span = self.span();
+        self.title
+            .get_ref(styles)
+            .clone()
+            .unwrap_or_else(|| {
+                Some(TextElem::packed(Packed::<BibliographyElem>::local_name_in(styles)))
+            })
+            .map(|title| {
+                HeadingElem::new(title)
+                    .with_depth(NonZeroUsize::ONE)
+                    .pack()
+                    .spanned(span)
+            })
+    }
+
+    /// Fetches and validates the bibliography works.
+    pub fn realize_works(
+        &self,
+        engine: &mut Engine,
+        styles: StyleChain,
+    ) -> SourceResult<std::sync::Arc<Works>> {
+        let span = self.span();
+        let works = Works::generate(engine).at(span)?;
+        if works.references.is_none() {
+            return Err(match self.style.get_ref(styles).source {
+                CslSource::Named(style, _) => eco_format!(
+                    "CSL style \"{}\" is not suitable for bibliographies",
+                    style.display_name()
+                ),
+                CslSource::Normal(..) => {
+                    "CSL style is not suitable for bibliographies".into()
+                }
+            })
+            .at(span);
+        }
+        Ok(works)
```

Comment:
Should do something like the original code here (to satisfy my other comment about guaranteeing references won't be `None`)


```rs
let references = works.references.as_ref().ok_or_else(|| {
    match self.style.get_ref(styles).source {
        CslSource::Named(style, _) => eco_format!(
            "CSL style \"{}\" is not suitable for bibliographies",
            style.display_name()
        ),
        CslSource::Normal(..) => {
            "CSL style is not suitable for bibliographies".into()
        }
    }
})
    .at(span)?;
Ok((works, references))
```